### PR TITLE
refactored `/challenges` routes and functions

### PR DESCRIPTION
**Pull request by swarajpure**
- Changed the base URL from `roadmap-site/challenges/` to `/challenges`
- Converted normal functions to arrow functions
- Changed how github username was being called

### CODE REVIEW

File: routes/challenges.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+const express = require('express')
+const router = express.Router()
+const authenticate = require('../middlewares/authenticate')
+const challengesController = require('../controllers/challengeController')
+
+/**
+ * @swagger
+ * /challenges/:
```
Comment (Rucha1499) [inline routes/challenges.js:8]: Any specific reason to add a ` / ` after ` /challenges ` ?  Got a bit confused here, rest great work!! 💯

File: controllers/challengeController.js [hunk 2]
```diff
@@ -38,24 +39,26 @@ const sendChallengeResponse = async (req, res) => {
   }
   return ''
```
Comment (whyDontI) [inline controllers/challengeController.js:40]: Unreachable return, if needed we can return from the `finally` block.

File: models/challenges.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+const Firestore = require('@google-cloud/firestore')
+const firestore = require('../utils/firestore')
+
+const challengesModel = firestore.collection('challenges')
+const userModel = firestore.collection('users')
+
+const cannotSubscribe = 'User cannot be subscribed to challenge'
+const userDoesNotExistError = 'User does not exist. Please register to participate'
+const errorMessage = 'Error getting challenges'
```
Comment (whyDontI) [inline models/challenges.js:14]: NIT: We should capitalize the constant variables

File: models/challenges.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    return (function () {
+      const challenges = []
+      challengesSnapshot.forEach((challengeDoc) => {
+        challenges.push({
+          id: challengeDoc.id,
+          ...challengeDoc.data()
+        })
+      })
+      return challenges
+    })()
```
Comment (whyDontI) [inline models/challenges.js:33]: Do we really need to use IIFE here?

**Review by whyDontI — CHANGES_REQUESTED** [review]
Thanks for these changes @swarajpure, Great work 👍🏽 
There is one doubt in the comments, please address that. Rest is Great 🎉

File: controllers/challengeController.js (hunk 2 above)
Comment (swarajpure) [reply controllers/challengeController.js:40]: Is it even needed?
The `catch` block will take care, right?

File: models/challenges.js (hunk 3 above)
Comment (swarajpure) [reply models/challenges.js:14]: Will do that.

File: models/challenges.js (hunk 4 above)
Comment (swarajpure) [reply models/challenges.js:33]: Even I was confused when I saw it 😅 
I guess I should just remove line number 24 and 31, right?

File: routes/challenges.js (hunk 1 above)
Comment (swarajpure) [reply routes/challenges.js:8]: I didn't add it, I just forgot to remove it 😅 
Thanks for informing though

Comment (Rucha1499) [reply routes/challenges.js:8]: okay! 👍

**Review by swarajpure — COMMENTED** [review]
@Rucha1499 @whyDontI Please have a look once. I have address the review comments.

**Review by whyDontI — DISMISSED** [review]
Great work 🎉 👍🏽

**Comment by Rucha1499** [comment]
Thanks for doing the changes! 👍

**Review by Rucha1499 — CHANGES_REQUESTED** [review]
Great work! 🎉  @swarajpure 

Requesting small changes. Please add the swagger docs for 5xx error. Found them missing. 😅

File: controllers/challengeController.js [hunk 5]
```diff
@@ -1,12 +1,13 @@
-const challengeQuery = require('../../models/roadmap-site/challenges')
-const subscribeToChallengeQuery = require('../../models/roadmap-site/subscribe-challenge')
+const challengeQuery = require('../models/challenges')
+
+const ERRORADMINSTRING = 'Something went wrong. Please try again or contact admin'
```
Comment (whyDontI) [inline controllers/challengeController.js:3]: It'll be great to use underscores for variable names.

File: models/challenges.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+const Firestore = require('@google-cloud/firestore')
+const firestore = require('../utils/firestore')
+
+const challengesModel = firestore.collection('challenges')
+const userModel = firestore.collection('users')
+
+const CANNOTSUBSCRIBE = 'User cannot be subscribed to challenge'
+const USERDOESNOTEXISTERROR = 'User does not exist. Please register to participate'
+const ERRORMESSAGE = 'Error getting challenges'
```
Comment (whyDontI) [inline models/challenges.js:14]: It'll be great to underscores in constant variable names.

**Review by whyDontI — CHANGES_REQUESTED** [review]
Great 👍🏽 
Requested some small changes.
Everything else looks good 👍🏽

**Review by Rucha1499 — APPROVED** [review]
Thank you for the quick fixes! @swarajpure 🎉 💯

**Review by whyDontI — APPROVED** [review]
🎉

File: public/apiSchema.json [hunk 7]
```diff
@@ -1,90 +1 @@
...
...rerAuth":[]}],"responses":{"200":{"description":"User details","content":{"application/json":{"schema":{"$ref":"#/components/schemas/users"}}}},"401":{"description":"unAuthorized","content":{"application/json":{"schema":{"$ref":"#/components/schemas/errors/unAuthorized"}}}},"404":{"description":"notFound","content":{"application/json":{"schema":{"$ref":"#/components/schemas/errors/notFound"}}}},"500":{"description":"badImplementation","content":{"application/json":{"schema":{"$ref":"#/components/schemas/errors/badImplementation"}}}}}}},"/users/:username":{"get":{"summary":"Get the details of user with provided id.","tags":["Users"],"security":[{"bearerAuth":[]}],"responses":{"200":{"description":"User details","content":{"application/json":{"schema":{"$ref":"#/components/schemas/users"}}}},"401":{"description":"unAuthorized","content":{"application/json":{"schema":{"$ref":"#/components/schemas/errors/unAuthorized"}}}},"404":{"description":"notFound","content":{"application/json":{"schema":{"$ref":"#/components/schemas/errors/notFound"}}}},"500":{"description":"badImplementation","content":{"application/json":{"schema":{"$ref":"#/components/schemas/errors/badImplementation"}}}}}}}}}
```
Comment (swarajpure) [inline public/apiSchema.json:1]: I didn't do anything on this file, the linter did 😅

**Review by swarajpure — COMMENTED** [review]
Clarifying regarding the changes in `apiSchema.json` file

**Review by ankurnarkhede — APPROVED** [review]
LGTM 👍

**Pull request merged**